syntax = "proto3";

package ficus;

import "pm_models.proto";

message GrpcContextKey {
    string name = 1;
}

message GrpcContextValue {
    oneof contextValue {
        string string = 1;
        GrpcHashesEventLogContextValue hashes_log = 2;
        GrpcNamesEventLogContextValue names_log = 3;
        uint32 uint32 = 4;
    }
}

message GrpcContextKeyValue {
    GrpcContextKey key = 1;
    GrpcContextValue value = 2;
}

message GrpcHashesEventLogContextValue {
    GrpcHashesEventLog log = 1;
}

message GrpcNamesEventLogContextValue {
    GrpcNamesEventLog log = 1;
}